velox/connectors/hive/iceberg/IcebergDataSink.cpp

```diff
@@ -0,0 +1,173 @@
+    facebook::velox::connector::CommitStrategy commitStrategy,
+    const std::shared_ptr<const HiveConfig>& hiveConfig)
+    : IcebergDataSink(
+          std::move(inputType),
+          insertTableHandle,
+          connectorQueryCtx,
+          commitStrategy,
+          hiveConfig,
+          [&insertTableHandle]() {
+            std::vector<column_index_t> channels;
+            const auto& inputColumns = insertTableHandle->inputColumns();
+
+            for (column_index_t i = 0; i < inputColumns.size(); i++) {
+              if (inputColumns[i]->isPartitionKey()) {
+                channels.push_back(i);
+              }
+            }
+            return channels;
+          }(),
+          [&insertTableHandle]() {
+            std::vector<column_index_t> channels(
+                insertTableHandle->inputColumns().size());
+            std::iota(channels.begin(), channels.end(), 0);
+            return channels;
+          }()) {}
+
+IcebergDataSink::IcebergDataSink(
+    RowTypePtr inputType,
+    std::shared_ptr<const HiveInsertTableHandle> insertTableHandle,
+    const ConnectorQueryCtx* connectorQueryCtx,
+    CommitStrategy commitStrategy,
+    const std::shared_ptr<const HiveConfig>& hiveConfig,
+    const std::vector<column_index_t>& partitionChannels,
+    const std::vector<column_index_t>& dataChannels)
+    : HiveDataSink(
+          inputType,
+          insertTableHandle,
+          connectorQueryCtx,
+          commitStrategy,
+          hiveConfig,
+          0,
+          nullptr,
+          partitionChannels,
+          dataChannels) {}
+
+std::vector<std::string> IcebergDataSink::close() {
+  setState(State::kClosed);
+  closeInternal();
+
+  auto icebergInsertTableHandle =
+      std::dynamic_pointer_cast<const IcebergInsertTableHandle>(
+          insertTableHandle_);
+
+  std::vector<std::string> commitTasks;
+  commitTasks.reserve(writerInfo_.size());
+  auto fileFormat = toString(insertTableHandle_->storageFormat());
+  std::string finalFileFormat(fileFormat);
+  std::transform(
+      finalFileFormat.begin(),
+      finalFileFormat.end(),
+      finalFileFormat.begin(),
+      ::toupper);
+
+  for (int i = 0; i < writerInfo_.size(); ++i) {
+    const auto& info = writerInfo_.at(i);
+    VELOX_CHECK_NOT_NULL(info);
+    auto commitDataJson = folly::toJson(
+        folly::dynamic::object(
+            "path",
+            info->writerParameters.writeDirectory() + "/" +
+                info->writerParameters.writeFileName())(
+            "fileSizeInBytes", ioStats_.at(i)->rawBytesWritten())(
+            "metrics",
+            folly::dynamic::object("recordCount", info->numWrittenRows))(
+            "partitionSpecId",
+            icebergInsertTableHandle->partitionSpec()->specId)(
+            "partitionDataJson",
+            partitionData_[i] == nullptr ? "" : partitionData_[i]->toJson())(
+            "fileFormat", finalFileFormat)("content", "DATA"));
+    commitTasks.push_back(commitDataJson);
```

Comment:
Use clang-format off to make the json readable, and comment the json format is consumed with Presto CommitTaskData